sql/path.cc

```diff
@@ -985,6 +987,25 @@ toUtf8String() const
     return result;
 }
 
+ssize_t
+Path::
+toIndex() const
+{
+    if (length_ != 1 || digits(0) != PathElement::DIGITS_ONLY)
+        return -1;
+    return at(0).toIndex();
```

Comment:
In this case, we know that we have length 1.  So toIndex() (which returns -1 if not) is the same as at(0).toIndex().